integration/spark2/src/main/scala/org/apache/spark/sql/hive/CarbonAnalysisRules.scala

```diff
@@ -95,12 +107,23 @@ case class CarbonIUDAnalysisRule(sparkSession: SparkSession) extends Rule[Logica
     def prepareTargetReleation(relation: UnresolvedRelation): SubqueryAlias = {
       val tupleId = UnresolvedAlias(Alias(UnresolvedFunction("getTupleId",
         Seq.empty, isDistinct = false), "tupleId")())
+
+      val localAlias = alias match {
+        case Some(a) => Some(alias.toSeq)
+        case _ => None
+      }
```

Comment:
Instead of use like this, you can use like `alias.map(Seq(_))` in all places

Reply:
Done